triqs/operators/many_body_operator.hpp

```diff
@@ -285,28 +285,32 @@ namespace operators {
    return res;
   }
 
-  // real
-  friend many_body_operator_generic real(many_body_operator_generic const& op) {
+  // Transform operator by applying a given function to each monomial
+  template <typename Lambda>
+  friend many_body_operator_generic foreach_monomial(many_body_operator_generic const& op, Lambda L) {
    many_body_operator_generic res;
-   using triqs::utility::real;
    using triqs::utility::is_zero;
    for (auto const& x : op) {
-    auto c = real(x.coef);
+    auto c = L(x.monomial, x.coef);
```

Comment:
What do you have in mind that might change the coeff depending on the monomial as well?